test/e2e/Apps/BasicDotNetIsolated/CustomExceptionPropertiesOrchestration.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Client;
+
+namespace Microsoft.Azure.Durable.Tests.E2E;
+
+public class CustomExceptionPropertiesOrchestration
+{
+    [Function(nameof(OrchestrationWithCustomException))]
+    public async Task<TaskFailureDetails?> OrchestrationWithCustomException([OrchestrationTrigger] TaskOrchestrationContext context)
+    {
+        // Call the activity that will throw an exception
+        try
+        {
+#pragma warning disable DURABLE2001 // Incorrect input type
+            await context.CallActivityAsync(nameof(BusinessActivity));
+#pragma warning restore DURABLE2001
+        }
+        catch (TaskFailedException ex)
+        {
+            return ex.FailureDetails;
+        }
+
+        // Should never reach here.
+        return null;
+
+    }
+
+    [Function(nameof(BusinessActivity))]
+    public void BusinessActivity([ActivityTrigger] TaskActivityContext context)
+    {
+        // Throw an exception with custom properties that should be captured
+        throw new BusinessValidationException(
+                message: "Business logic validation failed",
+                stringProperty: "validation-error-123",
+                intProperty: 100,
+                longProperty: 999999999L,
+                dateTimeProperty: new DateTime(2025, 10, 15, 14, 30, 0, DateTimeKind.Utc),
+                dictionaryProperty: new Dictionary<string, object?>
+                {
+                    ["error_code"] = "VALIDATION_FAILED",
+                    ["retry_count"] = 3,
+                    ["is_critical"] = true
+                },
+                listProperty: new List<object?> { "error1", "error2", 500, null },
+                nullProperty: null);
+    }
+
+    [Function("CustomExceptionProperties_HttpStart")]
+    public static async Task<HttpResponseData> CustomExceptionProperties_HttpStart(
+    [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestData req,
+    [DurableClient] DurableTaskClient client,
+    FunctionContext executionContext)
+    {
+        string instanceId = await client.ScheduleNewOrchestrationInstanceAsync(
+            nameof(OrchestrationWithCustomException));
+
+        return await client.CreateCheckStatusResponseAsync(req, instanceId);
+    }
```

Comment:
Is CustomExceptionProperties_HttpStart necessary, or can we use StartOrchestration instead?